config/routes.rb

```diff
@@ -3,170 +3,185 @@
+        get :art_pieces_per_day
+        get :favorites_per_day
+        get :user_visits_per_day
+        get :os_signups
+      end
 
-    resources :roles, except: [:show]
-    resources :cms_documents
-    resources :blacklist_domains, except: %i[show edit update]
-    resources :open_studios_events, only: %i[index edit new create update destroy] do
-      collection do
-        get :clear_cache
+      match '/discount/markup' => 'discount#markup', as: :discount_processor, via: %i[get post]
+
+      resources :roles, except: [:show]
+      resources :cms_documents
+      resources :blacklist_domains, except: %i[show edit update]
+      resources :open_studios_events, only: %i[index edit new create update destroy] do
+        collection do
+          get :clear_cache
+        end
       end
-    end
-    resources :email_lists, only: [:index] do
-      resources :emails, only: %i[index create new destroy]
-    end
-    resources :application_events, only: [:index]
-    resources :favorites, only: [:index]
-    resources :media, only: %i[index create new edit update destroy]
-    resources :artists, only: %i[index edit update] do
-      collection do
-        post :good_standing
-        post :bad_standing
-        post :pending
-        post :bulk_update, as: :bulk_update
-        get :purge
+      resources :email_lists, only: [:index] do
+        resources :emails, only: %i[index create new destroy]
       end
-      member do
-        post :suspend
+      resources :application_events, only: [:index]
+      resources :favorites, only: [:index]
+      resources :media, only: %i[index create new edit update destroy]
+      resources :artists, only: %i[index edit update] do
+        collection do
+          post :good_standing
+          post :bad_standing
+          post :pending
+          post :bulk_update, as: :bulk_update
+          get :purge
+        end
+        member do
+          post :suspend
+        end
       end
-    end
-    resources :users, only: [:show]
+      resources :users, only: [:show]
 
-    resources :art_piece_tags, only: %i[index destroy] do
-      collection do
-        get :cleanup
+      resources :art_piece_tags, only: %i[index destroy] do
+        collection do
+          get :cleanup
+        end
       end
-    end
 
-    resources :studios, only: %i[index new edit create update destroy] do
-      collection do
-        post :reorder
-      end
-      member do
-        post :unaffiliate_artist
+      resources :studios, only: %i[index new edit create update destroy] do
+        collection do
+          post :reorder
+        end
+        member do
+          post :unaffiliate_artist
+        end
       end
     end
+
+    get '/admin' => 'admin#index', as: :admin
+
+    get '/sitemap.xml' => 'main#sitemap', as: :sitemap
+
+    # legacy urls
+    get '/main/openstudios', to: redirect('/open_studios')
+    get '/openstudios', to: redirect('/open_studios')
+  end
+
+  # Routes available for all domains
```

Comment:
here we leave `admin/cms_documents` and `api` namespaces routes out of constraints so *everyone* can get to them.  All the rest are constrained to 1 or the other app.